src/libraries/System.Text.Json/tests/System.Text.Json.Tests/System.Text.Json.Tests.csproj

```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkMinimum)</TargetFrameworks>
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <!-- SYSLIB0020: JsonSerializerOptions.IgnoreNullValues is obsolete -->
     <NoWarn>$(NoWarn);SYSLIB0020</NoWarn>
```

Comment:
Tests validating the new behavior?

Reply:
I've added one larger test here https://github.com/krwq/runtime-1/pull/9/files#diff-9ca8a434d4ee452728961d1242d8680e5f4b72c8973a99b7ef59ecb005541025R131 exercising all trickier cases. Everything else should be covered by existing test cases